pkg/asset/agent/agentconfig/agenthosts.go

```diff
@@ -202,7 +202,8 @@ func (a *AgentHosts) validateRendezvousIPNotWorker(rendezvousIP string, hosts []
 	if rendezvousIP != "" {
 		for i, host := range hosts {
 			hostPath := field.NewPath("Hosts").Index(i)
-			if strings.Contains(string(host.NetworkConfig.Raw), rendezvousIP) && host.Role == workerRole {
+			ipString := fmt.Sprintf("ip: %s", rendezvousIP)
+			if strings.Contains(string(host.NetworkConfig.Raw), ipString) && host.Role == workerRole {
```

Comment:
I think it could be worth refactoring and extracting the already existing code in [GetFirstIP()](https://github.com/openshift/installer/blob/19c87c40a42acd92323c32b62af8de605498e5bb/pkg/asset/agent/manifests/nmstateconfig.go#L267-L288), so that to have a common util func like `GetNMStateInterfacesIP(nmstateRaw []byte)`, capable to return two lists (ipv5/ipv6) of the currently found IPs in the `interfaces` section of the given nmstate piece (also checking that they are really IP by using a `net.ParseIP`)
In this way such util func could be used both here - to verify that the configured rendezvousIP is not within a worker - and in nmstateconfig.go

Reply:
nit: for the happy path (and to avoid an unnecessary comparison):
```
if host.Role != workerRole {
  continue
}
```

Reply:
> I think it could be worth refactoring and extracting the already existing code in [GetFirstIP()](https://github.com/openshift/installer/blob/19c87c40a42acd92323c32b62af8de605498e5bb/pkg/asset/agent/manifests/nmstateconfig.go#L267-L288), so that to have a common util func like `GetNMStateInterfacesIP(nmstateRaw []byte)`,

Good suggestion. Moved the nmstate functions to common area and used  GetAllHostIPs() for check

Reply:
> nit: for the happy path (and to avoid an unnecessary comparison):
> 
> ```
> if host.Role != workerRole {
>   continue
> }
> ```

Done.